test/tools/jsontests/TransactionTests.cpp

```diff
@@ -38,134 +39,164 @@ namespace fs = boost::filesystem;
 
 namespace dev {  namespace test {
 
+mObject getExpectSection(mValue const& _expect, eth::Network _network)
+{
+	mObject obj;
+	for (auto const& value : _expect.get_array())
+	{
+		BOOST_REQUIRE(value.type() == json_spirit::obj_type);
+		obj = value.get_obj();
+		BOOST_REQUIRE_MESSAGE(obj.count("network"), "network section not set in expect section!");
+		vector<string> networks;
+		ImportTest::parseJsonStrValueIntoVector(obj["network"], networks);
+		BOOST_CHECK_MESSAGE(networks.size() > 0, TestOutputHelper::get().testName() + " Network array not set!");
+		ImportTest::checkAllowedNetwork(networks);
+
+		if (test::inArray(networks, test::netIdToString(_network)) || test::inArray(networks, string("ALL")))
+			return obj;
```

Comment:
Maybe just push back `obj` into a vector, and after the loop, check that the vector contains only one element.

Reply:
is it bad to have return in a loop ?

Reply:
Just in case many fields match.